track_syscalls/main.go

```diff
@@ -17,129 +20,170 @@ import (
 	"golang.org/x/sys/unix"
 )
 
-//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -type event ebpf bpf_program.c
-
-const (
-	tracepoint    = "raw_syscalls:sys_enter"
-	maxStackDepth = 20
-)
-
-type Allowlist struct {
-	Dependencies map[string][]int `json:"dependencies"`
-}
-
-type symbolInfo struct {
-	name  string
-	start uint64
-	end   uint64
-}
-
-var symbolCache []symbolInfo
+//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -type event ebpf backend.c
 
 func main() {
-
 	log.SetFlags(log.Ltime)
 
-	// Add a flag for the binary path
 	binaryPath := flag.String("binary", "", "Path to the binary for syscall tracking")
-	allowlistPath := flag.String("allowlist", "allowlist.json", "Path to the allowlist JSON file")
+	mode := flag.String("mode", "enforce", "Execution mode: 'build', 'enforce' or 'trace'")
 	flag.Parse()
 
-	// Check if the binary path is provided
 	if *binaryPath == "" {
 		log.Fatal("Please provide a binary path using the -binary flag")
 	}
-	if *allowlistPath == "" {
-		log.Fatal("Please provide a path to the allowlist JSON file using the -allowlist flag")
-	}
-	allowlist, err := loadAllowlist(*allowlistPath)
-	if err != nil {
-		log.Fatalf("loading allowlist: %v", err)
-	}
 
-	// Populate function symbols cache from the binary
-	if err := populateSymbolCache(*binaryPath); err != nil {
-		log.Fatalf("populating symbol cache: %v", err)
+	switch *mode {
+	case "build":
+		runBuildMode(*binaryPath)
```

Comment:
Let's rename this to `augment` because we take input as allowlist from capslock and we will augment it as we discussed [here](https://github.com/chains-project/goleash/pull/1#issuecomment-2399350141).

Reply:
I agree and i like this new name.